src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -4535,10 +4535,10 @@ void C2_MacroAssembler::vector_popcount_integral(BasicType bt, XMMRegister dst,
   }
 }
 
-
 void C2_MacroAssembler::vector_popcount_integral_evex(BasicType bt, XMMRegister dst, XMMRegister src,
                                                       KRegister mask, bool merge, int vec_enc) {
   assert(VM_Version::supports_avx512vl() || vec_enc == Assembler::AVX_512bit, "");
+  assert(UsePopCountInstruction, "");
```

Comment:
Please remove this assert on UsePopCountInstruction.

Reply:
Again, its counter intuitive to enter into this control flow with UsePopcountInstruction being false.